esphome/components/mcp3428/__init__.py

```diff
@@ -0,0 +1,33 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import i2c
+from esphome.const import CONF_ID
+
+CODEOWNERS = ["@mdop"]
+DEPENDENCIES = ["i2c"]
+MULTI_CONF = True
+
+mcp3428_ns = cg.esphome_ns.namespace("mcp3428")
+MCP3428Component = mcp3428_ns.class_("MCP3428Component", cg.Component, i2c.I2CDevice)
+
+CONF_CONTINUOUS_MODE = "continuous_mode"
+CONF_MCP3428_ID = "mcp3428_id"
+
+CONFIG_SCHEMA = (
+    cv.Schema(
+        {
+            cv.GenerateID(): cv.declare_id(MCP3428Component),
+            cv.Optional(CONF_CONTINUOUS_MODE, default=False): cv.boolean,
+        }
+    )
+    .extend(cv.COMPONENT_SCHEMA)
+    .extend(i2c.i2c_device_schema(None))
```

Comment:
Devices with address pins still have a default address if the pins are left floating.
```suggestion
    .extend(i2c.i2c_device_schema(0x58))
```

Reply:
Ok, but left floating the address is 0x68, like with grounded pins. Addressed it in a separate commit.